Library/Homebrew/dev-cmd/pull.rb

```diff
@@ -102,6 +102,18 @@ def pull
       ENV["GIT_COMMITTER_EMAIL"] = ENV["HOMEBREW_GIT_EMAIL"]
     end
 
+    # Depending on user configuration, git may try to invoke gpg.
```

Comment:
Is there any way to identify this in advance rather than doing so unconditionally?

Reply:
```ruby
Utils.popen_read("git config --get commit.gpgsign").chomp
```
Maybe?

Reply:
@alyssais any thoughts?

Reply:
What @DomT4 says looks good, though I'm not sure there's any harm in doing it unconditionally. I'll try to push that change tonight if we want it.

Reply:
Yeh, that would be good, thanks 👍